hr_attendance_reason/static/src/xml/attendance.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+    <t t-extend="HrAttendanceMyMainMenu">
+        <t t-jquery="h4:first" t-operation="after">
+            <t t-if="widget.attendance_reasons_in_kiosk">
+                <div class="reasons_lists" t-if="widget.reasons">
+                    <h3 class="mt8">Reason</h3>
+                    <select
+                        class="o_hr_attendance_reason o_input_dropdown o_input o_field_widget o_required_modifier"
+                    >
+                        <option value="0" />
+                        <t t-foreach="widget.reasons" t-as="reason">
```

Comment:
Although maybe you can do it in one go here. Somethin like this (not tested):
```suggestion
                        <t t-foreach="widget.reasons.filter(reason => {return (checked_in && reason.action_type === 'sign_out') || (!checked_in && reason.action_type === 'sign_in')})" t-as="reason">
```

Reply:
This and similar things don't work, I don't know exactly the syntax to use in this use case, sorry.

Reply:
Looks like the compiler messes the lambda arguments up. Not much to worry about anyway :)